052-cruise-control-api-users.md

```diff
@@ -0,0 +1,204 @@
+# Add ability to create Cruise Control REST API users
+
+This proposal is about giving developers the ability to create REST API users for the Cruise Control REST API.
+This would allow roles and permissions to be defined to allow developers and third-party applications to access the Cruise Control REST API without having to disable HTTP basic authentication.
+
+## Current situation
+
+At this time, Strimzi creates two Cruise Control API users:
+
+* `admin`      - Used by Strimzi for Cruise Control admin operations.
+* `heathcheck` - Used by Strimzi for checking the readiness of the Cruise Control application.
+
+The access to these API users are limited to Strimzi.
+So when HTTP basic authentication is enabled for Cruise Control, no one can access the the Cruise Control API other than Strimzi.
+
+## Motivation
+
+There are a couple use cases where developers would want to access the Cruise Control API without disabling API security:
+
+* Debugging Cruise Control in a production environment
+* Running the Cruise Control UI alongside a Strimzi-managed Kafka cluster 
+
+In order to allow developers and third-party applications to access the Cruise Control API without needing to disable API security, we would need to provide a method of creating API users.
```

Comment:
Again, why would they want/need it? What will they be able to do over the API?

Reply:
As others say, if we enable admin operations for third-party, there will be a race condition between Strimzi and these applications, much like the race condition we have today between CC and KafkaRoller. The big difference is that we don't control the third-party applications.

Reply:
Agreed, just added some more justifications in recent changed, let me know what you think!